hmp/models/base.py

```diff
@@ -0,0 +1,205 @@
+"""Models to estimate event probabilities."""
+import gc
+import itertools
+import multiprocessing as mp
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from itertools import cycle, product
+from typing import Any
+from warnings import resetwarnings, warn
+
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from pandas import MultiIndex
+from scipy.signal import correlate
+from scipy.stats import norm as norm_pval
+
+from hmp.trialdata import TrialData
+
+try:
+    __IPYTHON__
+    from tqdm.notebook import tqdm
+except NameError:
+    from tqdm import tqdm
+
+
+@dataclass
+class EventProperties():
+    sfreq: float
+    steps: float
+    shape: float
+    width: int
+    width_samples: int
+    location: int
+    template: Any
+
+    @classmethod
+    def from_standard_data(cls, data, sfreq=None, shape=2, width=50, template=None, location=None):
```

Comment:
While the EventProperties class is a very good structure, I think this is a bit confusing 
```suggestion
    def expected_event(cls, data, sfreq=None, shape=2, width=50, template=None, location=None):
```

Reply:
Additionally, I think giving the data is a bit confusing as this is really only about the events we except in the timeseries. In fact we only need the sampling frequency. So rather than asking participant to provide the data we just ask for the sampling frequency.

Reply:
I have renamed it create_expected, does that make sense?